src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -126,6 +134,89 @@ public void deleteSession(Session target) {
     }
 
+    /**
+     * Assigns the student/tutor to a session in addressBook
+     * @param assignment
+     */
+    public void registerAssignment(Assignment assignment) throws TimeClashException, AlreadyEnrolledException,
+            SessionNotFoundException, StudentTutorNotFoundException {
+        Session session = addressBook.getSession(assignment.getSessionId());
+        if (session == null) {
+            throw new SessionNotFoundException("Session id :" + assignment.getSessionId() + " does not exist");
+        }
+        if (assignment.getStudentId() != null) {
+            Person currentStudent = addressBook.getPerson(assignment.getStudentId());
+            if (currentStudent == null) {
+                throw new StudentTutorNotFoundException("student id: " + assignment.getStudentId() + " does not exist");
+            }
+            if (!checkEnrollement(currentStudent, session)) {
+                if (checkTimeSlot(currentStudent, session)) {
+                    currentStudent.addSession(assignment.getSessionId());
+                    session.assignStudent(assignment.getStudentId());
+                } else {
+                    throw new TimeClashException("There is timing clash between "
+                            + "the student sessions and session to be added!");
+                }
+            } else {
+                throw new AlreadyEnrolledException("The student of student Id: "
+                        + currentStudent.getPersonId().getPersonId()
+                        + " is already enrolled in this class");
+            }
+        }
+        if (assignment.getTutorId() != null) {
+            Person currentTutor = addressBook.getPerson(assignment.getTutorId());
+            if (currentTutor == null) {
+                throw new StudentTutorNotFoundException("tutor id: " + assignment.getTutorId() + " does not exist");
+            }
+            if (!checkEnrollement(currentTutor, session)) {
+                if (checkTimeSlot(currentTutor, session)) {
+                    currentTutor.addSession(assignment.getSessionId());
+                    session.assignTutor(assignment.getTutorId());
+                } else {
+                    throw new TimeClashException("There is timing clash between "
+                            + "the tutor sessions and session to be added!");
+                }
+            } else {
+                throw new AlreadyEnrolledException("The student of tutor Id: "
+                        + currentTutor.getPersonId().getPersonId()
+                        + " is already assigned to this class");
+            }
+        }
+    }
+
+    private boolean checkEnrollement(Person person, Session session) {
+        for (SessionId sessionid : person.getSessions()) {
+            if (sessionid.equals(session.getClassId())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean checkTimeSlot(Person person, Session session) {
```

Comment:
This method i think can be abstracted too. Can call a method in person which calls a method in the session list.